### PR TITLE
C_CPP:Function overloading concepts c++ #1508

### PR DESCRIPTION
**Pull request by subhoshreep**
<hr>

## Description 📜

This is documentation on concepts of function overloading in c++. 
contributed by :
subhoshree pattnaik
batch 3 cpp
Fixes # <1508> 

<hr>

## Type of change 📝

- [x] Doc#umentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by subhoshreep** [comment]
@Suvraneel can you please review my PR
subhoshree pattnaik
batch 3 cpp

**Comment by Suvraneel** [comment]
@kashishahuja2002 @apeksha235

File: C_CPP/Functions & Recursion and its types/function overloading.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+   cout<<"\n principal amt: "<<pr; 
+    cout<<"\n time : "<<t<<"years"; 
+    cout<<"\n rate : "<<r; 
+    cout<<"\n Simple interest : "<<(pr*t*r)<<"\n"; 
+}
+int main() 
+{
+    system("cls"); 
+    cout<<"case1"; 
+    amt(1000);
```
Comment (apeksha235) [inline C_CPP/Functions & Recursion and its types/function overloading.md:49]: Add comments to explain the code

**Review by apeksha235 — CHANGES_REQUESTED** [review]
Add comments to explain your code.. 
Also add a line to the Functions & Recursion and its types/readme.md
@subhoshreep

File: C_CPP/Functions & Recursion and its types/readme.md [hunk 2]
```diff
@@ -1 +1 @@
-
+### 1.FUNCTION OVERLOADING CONCEPT
```
Comment (vasu-1) [inline C_CPP/Functions & Recursion and its types/readme.md:1]:
```suggestion
-  [FUNCTION OVERLOADING CONCEPT](function%20overloading.md)
```

File: C_CPP/Functions & Recursion and its types/function overloading.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# FUNCTION OVERLOADING
+
+### DEFINATION- A function name having several definitions that are differentiable by the number or types of their arguments is known as function overloading.
+## Why we need it? How is it useful?
```
Comment (vasu-1) [inline C_CPP/Functions & Recursion and its types/function overloading.md:4]:
```suggestion
- **Definition** : A function name having several definitions that are differentiable by the number or types of their arguments is known as function overloading.
### Why we need it? How is it useful?
```

File: C_CPP/Functions & Recursion and its types/function overloading.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+b. A match through promotion<br>
+c. A match through application of standard c++ conversion rules.<br>
+d. A match through application of a user defined conversion.<br></p>
+
+## What are the advantages of function overloading?
+<p>By overloading function we tend to save compiler from the trouble of pushing default argument values on the function call stack, hence the function does not require to test default values.
+Also a class can also have overloaded constructors. A destructor function can’t be overloaded.</p>
+
+## Example of function overloading with help of a program on calculating simple interest with default arguments.
+```
```
Comment (vasu-1) [inline C_CPP/Functions & Recursion and its types/function overloading.md:34]:
```suggestion

### Example of function overloading with help of a program on calculating simple interest with default arguments.
```c++
```

File: C_CPP/Functions & Recursion and its types/function overloading.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+## How to declare or implement function overloading?
+<p>The key to function overloading is a function’s argument list which is also known as the function signature.
+There might be a situation where a function has same name and same number of arguments then we must keep in mind that c++ lets you overload a function name only if the function has same name but different signatures.<br>
+Example-<br>
+Void area (int l, int b);<br>
+Void area (int r);<br>
+Note-Also if the function has same name and different signature with different return types are allowed.</p>
+
+## How to call an overloaded function?
```
Comment (vasu-1) [inline C_CPP/Functions & Recursion and its types/function overloading.md:15]:
```suggestion
### How to call an overloaded function?
```

File: C_CPP/Functions & Recursion and its types/function overloading.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+3.	Ambiguous match <br>
+   <br>
+So in order to deal with this the compiler follows the following steps to find the best match: <br>
+   <br>
+a. Search for an exact match by comparing the type of the argument with the defined instance type.<br>
+b. A match through promotion<br>
+c. A match through application of standard c++ conversion rules.<br>
+d. A match through application of a user defined conversion.<br></p>
+
+## What are the advantages of function overloading?
```
Comment (vasu-1) [inline C_CPP/Functions & Recursion and its types/function overloading.md:29]:
```suggestion
### What are the advantages of function overloading?
```

File: C_CPP/Functions & Recursion and its types/function overloading.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
+# FUNCTION OVERLOADING
+
+### DEFINATION- A function name having several definitions that are differentiable by the number or types of their arguments is known as function overloading.
+## Why we need it? How is it useful?
+<p>Function overloading not just implements polymorphism but also reduces number of comparisons in a program and hence makes the program faster.</p>
+
+## How to declare or implement function overloading?
```
Comment (vasu-1) [inline C_CPP/Functions & Recursion and its types/function overloading.md:7]:
```suggestion
### How to declare or implement function overloading?
```

File: C_CPP/Functions & Recursion and its types/function overloading.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    cout<<"case3"; 
+    amt(1000,3,0.11); //three arguments are passed
+   cout<<"case4";
+    amt(1000,0.12); //two arguments passed with t=2 as default
+    return 0; 
+} 
+ ``` 
+### Output of the program:
+ ```
+ case1
```
Comment (vasu-1) [inline C_CPP/Functions & Recursion and its types/function overloading.md:61]: Here at line 58 and 60
remove that one white space at the beginning of the line.

**Review by vasu-1 — CHANGES_REQUESTED** [review]
some syntax changes are required !
Please do it !

    ** For now it is okk but get rid of html tags in .md files !!!! **

File: C_CPP/Functions & Recursion and its types/function overloading.md (hunk 8 above)
Comment (subhoshreep) [reply C_CPP/Functions & Recursion and its types/function overloading.md]: done

**Comment by subhoshreep** [comment]
how can i resolve this conflict problem?

**Comment by subhoshreep** [comment]
https://github.com/girlscript/winter-of-contributing/pull/2551
I have created another pr to resolve the branch conflict problem.
please review that PR of mine.
sorry for causing the ambiguity.

**Comment by vasu-1** [comment]
kindly close this PR ASAP... :))

**Pull request closed without merging**